source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -3707,6 +3707,39 @@ void Graph::searchNodePopup(bool cursor)
     }
 }
 
+bool Graph::isPinHovered() {
+    ed::PinId currentPin = ed::GetHoveredPin();
+    ed::PinId nullPin = 0;
+    return currentPin != nullPin;
+}
+
+void Graph::addPinPopup() {
+    // Add a floating popup to pin when hovered
+    if (isPinHovered()) {
+        ed::Suspend();
+        UiPinPtr pin = getPin(ed::GetHoveredPin());
+        std::string connected = "";
+        std::string value = "";
+        if (pin->_connected) {
+            connected = "\nConnected to";
+            if (pin->_name == "out") {
```

Comment:
I think this may miss some outputs as not all output pins are called "out". e.g. UsdUvTexture, the outputs are called "r", "g" , "b" , "a" , "rgb" and "rgba". 

Also the else case, you can't assume that the input pin is connected to an output pin called "out".

Reply:
Okay, I added some connections back and removed that if statement so now it works for both input and output pins